src/mono/mono/component/marshal-ilgen.h

```diff
@@ -19,21 +36,20 @@ typedef struct {
 	int (*emit_marshal_custom) (EmitMarshalContext *m, int argnum, MonoType *t, MonoMarshalSpec *spec, int conv_arg, MonoType **conv_arg_type, MarshalAction action);
 	int (*emit_marshal_asany) (EmitMarshalContext *m, int argnum, MonoType *t, MonoMarshalSpec *spec, int conv_arg, MonoType **conv_arg_type, MarshalAction action);
 	int (*emit_marshal_handleref) (EmitMarshalContext *m, int argnum, MonoType *t, MonoMarshalSpec *spec, int conv_arg, MonoType **conv_arg_type, MarshalAction action);
-} MonoMarshalIlgenCallbacks;
+} MonoMarshalILgenCallbacks;
+
+MONO_COMPONENT_EXPORT_ENTRYPOINT
+MonoComponentMarshalILgen* mono_component_marshal_ilgen_init (void);
 
 void
-mono_install_marshal_callbacks_ilgen (MonoMarshalIlgenCallbacks *cb);
+mono_install_marshal_callbacks_ilgen (MonoMarshalILgenCallbacks *cb);
 
 
-MONO_API void
-mono_marshal_ilgen_init (void);
- 
-void
-mono_marshal_noilgen_init_heavyweight (void);
 
 void
-mono_marshal_noilgen_init_lightweight (void);
+mono_marshal_ilgen_init_internal (void);
 
+ 
 int
 mono_emit_marshal_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 	      MonoMarshalSpec *spec, int conv_arg,
```

Comment:
This function and `mono_marshal_ilgen_init_internal` are no longer called directly. so they could be static in marshal-ilgen.c and these declarations can be removed from the header file.